Components/Helper/Image.php

```diff
@@ -183,15 +187,24 @@ public static function getDetailImageUrl(Detail $detail)
             ->Models()
             ->getRepository('Shopware\Models\Article\Image')
             ->findOneBy(array('articleDetail' => $detail));
-
-        if (!is_null($detailImage)) {
-            $imagePath = $detailImage->getParent()->getMedia()->getPath();
-            return $mediaService->getUrl($imagePath);
+        if ($detailImage) {
+            try {
+                $imagePath = $detailImage->getParent()->getMedia()->getPath();
+                return $mediaService->getUrl($imagePath);
+            } catch (\Exception $e) {
+                Shopware()->Plugins()->Frontend()->NostoTagging()->getLogger()->error($e->getMessage());
+            }
+            try {
+                // Fallback to article main image
+                if (!empty($detail->getArticle()->getImages()->first()->getMedia())) {
```

Comment:
What happens if there are no images in the Article? In other words if `$detail->getArticle()->getImages()` returns an empty collection?

Reply:
We can't have a detail without having images into the article itself, since detail images comes from the parent article. So the first if condition in the line 190 would be skipped and return null.
However i'll add an extra check just in case